vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -478,16 +466,12 @@ func (e *StandardTxExecutor) AddPermissionlessDelegatorTx(tx *txs.AddPermissionl
 		return err
 	}
 
-	txID := e.Tx.ID()
-	newStaker, err := state.NewPendingStaker(txID, tx)
-	if err != nil {
+	if err := e.addStakerFromStakerTx(tx); err != nil {
 		return err
 	}
 
-	e.State.PutPendingDelegator(newStaker)
 	avax.Consume(e.State, tx.Ins)
-	avax.Produce(e.State, txID, tx.Outs)
-
+	avax.Produce(e.State, e.Tx.ID(), tx.Outs)
```

Comment:
can we hoist `e.Tx.ID()`? we do this in other functions in this file and I think it reads better

```golang
	txID := e.Tx.ID()
	avax.Consume(e.State, tx.Ins)
	avax.Produce(e.State, txID, tx.Outs)
```

Reply:
Done